cdc/owner/owner.go

```diff
@@ -599,27 +599,45 @@ func (o *ownerImpl) handleQueries(query *Query) error {
 		}
 		query.Data = ret
 	case QueryHealth:
-		isHealthy := true
-		if !o.changefeedTicked {
-			// Owner has not yet tick changefeeds, some changefeeds may be not
-			// initialized.
-			isHealthy = false
-		} else {
-			for _, cfReactor := range o.changefeeds {
-				provider := cfReactor.GetInfoProvider()
-				if provider == nil || !provider.IsInitialized() {
-					// The scheduler has not been initialized yet, it is considered
-					// unhealthy, because owner can not schedule tables for now.
-					isHealthy = false
-					break
-				}
-			}
+		isHealthy, err := o.isHealthy(ctx)
+		if err != nil {
+			return errors.Trace(err)
 		}
 		query.Data = isHealthy
 	}
 	return nil
 }
 
+func (o *ownerImpl) isHealthy(ctx context.Context) (bool, error) {
+	if !o.clusterVersionConsistent(o.captures) {
+		return false, nil
+	}
+
+	upstreamM, err := o.upstreamManager.GetDefaultUpstream()
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+	if err := version.CheckStoreVersion(ctx, upstreamM.PDClient, 0); err != nil {
```

Comment:
do we need to check version consistency for every health API call?

Reply:
Some API works even though the cluster is not consistent, but this can not work properly.

If we try to make every API check version, we have to investigate each one first.